src/ModelContextProtocol/Protocol/Transport/TransportBase.cs

```diff
@@ -76,6 +76,9 @@ protected async Task WriteMessageAsync(JsonRpcMessage message, CancellationToken
             throw new InvalidOperationException("Transport is not connected");
         }
 
+        var messageId = (message as JsonRpcMessageWithId)?.Id.ToString() ?? "(no id)";
```

Comment:
Now that Id.ToString is likely to allocate, after your change to make it typically number rather than string based, we should guard this with an IsEnabled check, right?